server/src/main/java/org/opensearch/indices/replication/SegmentReplicationState.java

```diff
@@ -60,23 +66,42 @@ public static Stage fromId(byte id) {
 
     private Stage stage;
     private final ReplicationLuceneIndex index;
-    private final ReplicationTimer timer;
+    private final ReplicationTimer overallTimer;
+    private final ReplicationTimer stepTimer;
```

Comment:
Very minor, but I think logically `stageTimer` is better name (since we deal with stages and not steps)